### PR TITLE
[Merged by Bors] - feat: using UnivLE in constructing limits in `Type`

### PR DESCRIPTION
**Pull request by kim-em**
---

- [x] depends on: #5723

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#5723
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/CategoryTheory/Limits/Shapes/Types.lean [hunk 1]
```diff
@@ -64,6 +71,13 @@ theorem pi_lift_π_apply {β : Type v} (f : β → TypeMax.{v, u}) {P : TypeMax.
...
+/-- A variant of `pi_lift_π_apply` using `UnivLE` rather than `TypeMax`. -/
+@[simp 1001]
+theorem UnivLE.pi_lift_π_apply [UnivLE.{v, u}] {β : Type v} (f : β → Type u) {P : Type u}
+    (s : ∀ b, P ⟶ f b) (b : β) (x : P) :
+    (Pi.π f b : (piObj f) → f b) (@Pi.lift β _ _ f _ P s x) = s b x :=
+  congr_fun (limit.lift_π (Fan.mk P s) ⟨b⟩) x
+
 /-- A restatement of `Types.Limit.lift_π_apply` that uses `Pi.π` and `Pi.lift`,
 with specialized universes. -/
 @[simp 1001]
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/Shapes/Types.lean:83]:
```suggestion
```

File: Mathlib/CategoryTheory/Limits/Shapes/Types.lean [hunk 2]
```diff
@@ -73,13 +87,21 @@ theorem pi_lift_π_apply' {β : Type v} (f : β → Type v) {P : Type v}
...
+/-- A variant of `pi_map_π_apply` using `UnivLE` rather than `TypeMax`. -/
+@[simp 1001]
+theorem UnivLE.pi_map_π_apply [UnivLE.{v, u}] {β : Type v} {f g : β → Type u}
+    (α : ∀ j, f j ⟶ g j) (b : β) (x) :
+    (Pi.π g b : ∏ g → g b) (Pi.map α x) = α b ((Pi.π f b : ∏ f → f b) x) :=
+  Limit.map_π_apply.{v, u} _ _ _
+
 /-- A restatement of `Types.Limit.map_π_apply` that uses `Pi.π` and `Pi.map`,
 with specialized universes. -/
 @[simp 1001]
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/Shapes/Types.lean:107]:
```suggestion
```

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#5723~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/CategoryTheory/Limits/Shapes/Types.lean [hunk 3]
```diff
@@ -556,6 +595,12 @@ end Cofork
 
 section Pullback
 
+-- #synth HasPullbacks.{u} (Type u)
+instance : HasPullbacks.{u} (Type u) :=
+  -- FIXME does not work via `inferInstance` despite `#synth HasPullbacks.{u} (Type u)` succeeding.
+  -- inferInstance
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/Shapes/Types.lean:601]: Please create an issue for this problem, and link to it.

File: Mathlib/CategoryTheory/Limits/Shapes/Types.lean [hunk 4]
```diff
@@ -639,4 +684,14 @@ theorem pullbackIsoPullback_inv_snd :
 
 end Pullback
 
+section Pushout
+
+-- #synth HasPushouts.{u} (Type u)
+instance : HasPushouts.{u} (Type u) :=
+  -- FIXME does not work via `inferInstance` despite `#synth HasPushouts.{u} (Type u)` succeeding.
+  -- inferInstance
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/Shapes/Types.lean:692]: Idem

**Comment by jcommelin** [comment]
bors d+

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**